lib/pages/product_page.dart

```diff
@@ -0,0 +1,126 @@
+      builder: (BuildContext context, AsyncSnapshot snapshot) {
+        if (snapshot.connectionState == ConnectionState.done) {
+          return Scaffold(
+            appBar: CustomAppBar(
+              title: '${product['name']}',
+              trailing: canEdit
+                  ? IconButton(
+                      icon: const Icon(Icons.edit),
+                      onPressed: () {
+                        // Edit the product
+                      },
+                    )
+                  : const SizedBox(),
+            ),
+            drawer: const CustomDrawer(),
+            body: Column(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Card(
+                    child: ListView(
+                      shrinkWrap: true,
+                      children: [
+                        const ListTile(
+                          title: Text(
+                            'Details:',
+                            style: TextStyle(
+                              fontWeight: FontWeight.bold,
+                              fontSize: 20,
+                            ),
+                          ),
+                        ),
+                        ListTile(
+                          title: Text('Category: ${product['category']}'),
+                        ),
+                        ListTile(
+                          title: Text('Barcode: ${product['barcode']}'),
+                        ),
+                        ListTile(
+                          title: Text(
+                              'Content: ${product['content']} ${product['unit']}'),
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Card(
+                    child: Column(
+                      children: [
+                        const ListTile(
+                          title: Text(
+                            'Locations:',
+                            style: TextStyle(
+                              fontWeight: FontWeight.bold,
+                              fontSize: 20,
+                            ),
+                          ),
+                        ),
+                        ListView.builder(
+                          shrinkWrap: true,
+                          itemCount: product['locations'].length,
+                          itemBuilder: (BuildContext context, int index) {
+                            return ListTile(
+                              title: Text(
+                                  '${product['locations'][index]['address']}'),
+                              subtitle: Text(
+                                  'Stock: ${product['locations'][index]['pivot']['stock']} Shelf: ${product['locations'][index]['pivot']['shelf_amount']}'),
+                            );
```

Comment:
Low priority but maybe reduce padding, add a divider between the locations and make the location bold?